plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -154,7 +105,11 @@ fullyApplyAndBetaReduce info args0 = do
               (\n -> if n == param then Just <$> PLC.rename arg else pure Nothing)
               acc
           go acc' arity' args'
-        _ -> pure Nothing
+        -- term/type argument mismatch, don't inline
+        (TermParam _:_, TypeAppContext{}) -> pure Nothing
+        (TypeParam _:_, TermAppContext{}) -> pure Nothing
+        -- no more arguments to apply, just apply what we have
+        (_, AppContextEnd) -> pure . Just $ fillAppContext acc args
```

Comment:
This is wrong. `args` is `AppContextEnd` here, so `fillAppContext acc args` does nothing - it just returns `acc`.